src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/HotSpotResolvedJavaMethodImpl.java

```diff
@@ -573,6 +573,19 @@ public boolean isDefault() {
         return ((getModifiers() & mask) == Modifier.PUBLIC) && getDeclaringClass().isInterface();
     }
 
+    /*
+     * Currently in hotspot an instance method can either be a "normal" or an "overpass" method.
+     * Overpass methods are instance methods which are created when otherwise a valid candidate
+     * for method resolution would not be found.
+     */
+    @Override
+    public boolean isDeclared() {
+        if (isConstructor() || isStatic()) {
+            return false;
+        }
+        return !compilerToVM().isOverpass(this);
```

Comment:
I think you can do this with a direct flag check:
```
boolean isOverpass = (getConstMethodFlags() & config().constMethodIsOverpass) != 0;
return isOverpass;
```
See #20256 as an example of the other changes needed for this.

Reply:
good call. changed